platform/include/platform/Span.h

```diff
@@ -320,6 +321,44 @@ struct Span {
     Span(element_type (&elements)[Extent]):
         _data(elements) { }
 
+    /**
+     * Construct a Span from the reference to an std::array.
+     *
+     * @param array Reference to the array viewed.
+     *
+     * @post a call to size() returns Extent, and data() returns a
+     * pointer to elements as if by @p array.data().
+     */
+    template<typename OtherElementType>
+    Span(std::array<OtherElementType, Extent> &array):
+        _data(array.data())
+    {
+        MBED_STATIC_ASSERT(
+            (span_detail::is_convertible<OtherElementType (*)[1], ElementType (*)[1]>::value),
+            "OtherElementType(*)[] should be convertible to ElementType (*)[]"
+        );
+    }
+
+    /**
+     * Construct a Span from the reference to a const std::array.
+     *
+     * @param array Const reference to the array viewed.
+     *
+     * @post a call to size() returns Extent, and data() returns a
+     * pointer to elements as if by @p array.data().
+     *
+     * @note ElementType must be const to accept a const std::array.
+     */
+    template<typename OtherElementType>
+    Span(const std::array<OtherElementType, Extent> &array):
+        _data(array.data())
+    {
+        MBED_STATIC_ASSERT(
+            (span_detail::is_convertible<OtherElementType (*)[1], ElementType (*)[1]>::value),
```

Comment:
You can now rely on the `std` type traits stuff, no need for the `span_detail` any more.